src/libtcod/bresenham.h

```diff
@@ -80,7 +80,67 @@ TCODLIB_API bool TCOD_line_step_mt(int* __restrict xCur, int* __restrict yCur, T
 TCOD_DEPRECATED("Use TCOD_line instead.")
 TCODLIB_API bool TCOD_line_mt(
     int xFrom, int yFrom, int xTo, int yTo, TCOD_line_listener_t listener, TCOD_bresenham_data_t* data);
+
 #ifdef __cplusplus
 }
-#endif
-#endif
+
+#include <functional>
+#include <vector>
+
+namespace tcod {
+/**
+tcod::Line class
+
+Implements a lightweight bresenham line drawing algorithm.
+ */
+class TCODLIB_API Line {
```

Comment:
Just `Line` might be to generic.  I might call this class `Bresenham` or `BresenhamLine` instead.